Don't assume the zlib version has 3 components: zlib 1.3 violates this.

Index: config/auto/zlib.pm
--- config/auto/zlib.pm.orig
+++ config/auto/zlib.pm
@@ -65,7 +65,7 @@ sub runstep {
 sub _evaluate_cc_run {
     my $self = shift;
     my ($conf, $test, $has_zlib) = @_;
-    if ( $test =~ m/^(\d\.\d\.\d)/ ) {
+    if ( $test =~ m/^(\d\.\d)/ ) {
         my $version = $1;
         $has_zlib = 1;
         $conf->debug(" (yes) ");
